deploy/deploy/aws/instances.py

```diff
@@ -0,0 +1,279 @@
+import boto3
+
+from rich import print
+from typing import Optional
+import time
+import typer
+from deploy.aws.config import AwsDeployConfig
+from deploy.aws import constants
+from pathlib import Path
+import requests
+from rich.console import Console
+import subprocess
+import os
+
+CONFIG_FILE_PATH = Path(__file__).parent / "config.yaml"
+if constants.CONFIG_PATH_OVERRIDE_ENVVAR in os.environ:
+    CONFIG_FILE_PATH = Path(os.environ[constants.CONFIG_PATH_OVERRIDE_ENVVAR])
+config = AwsDeployConfig.from_yaml_file(CONFIG_FILE_PATH)
+
+# TODO: Move these somewhere else
+HEALTHCHECK_PATIENCE_SECS = 60 * 5
+DEFAULT_TTL = 60 * 60 * 5  # 5 hours
+
+app = typer.Typer()
+
+
+def get_log_lines(host: str) -> list[str]:
+    out = subprocess.check_output(
+        f"ssh ubuntu@{host} cat /var/log/cloud-init-output.log",
+        shell=True,
+        stderr=subprocess.DEVNULL,
+    )
+    lines = out.decode("utf-8").splitlines()
+    return [l for l in lines if l.strip() != ""]
+
+
+def _launch(
+        instance_type: str,
+        checkout: str,
+        ttl_secs: int,
+        wait,
+        idempotency_token: Optional[str] = None,
+):
+    # Validate config
+    print()
+    print("Checking config fully set...")
+    try:
+        config.validate_full_config_set()
+    except ValueError as e:
+        print(e)
+        print("Not all config values are set. Did you create the security group and IAM resources?")
+        return
+    print("[green]✓[/green] Config fully set")
+    print()
+
+    # Get appropriate AMI for architecture
+    if instance_type not in constants.INSTANCE_TO_AMI:
+        raise ValueError(f"Instance type {instance_type} architecture not currently known (add it to "
+                         f"INSTANCE_TO_AMI in constants.py)")
+    ami = constants.INSTANCE_TO_AMI[instance_type]
+
```

Comment:
Add check that we don't support non-ARM AMIs at the moment due to the docker-compose installation script hardcoded ARM